src/run.rs

```diff
@@ -45,772 +50,31 @@ fn spin_loop() {} // this could use `std::hint::spin_loop`, but in practice it h
-  /// let rgt = net.do_op1(lab, num, trg);
-  /// ```
-  Op1 { op: Op, num: u64, trg: TrgId, rgt: TrgId },
-  /// See [`Net::do_mat`].
-  /// ```rust,ignore
-  /// let (lft, rgt) = net.do_mat(trg);
-  /// ```
-  Mat { trg: TrgId, lft: TrgId, rgt: TrgId },
-  /// See [`Net::do_wires`].
-  /// ```rust,ignore
-  /// let (av, aw, bv, bw) = net.do_wires();
-  /// ```
-  Wires { av: TrgId, aw: TrgId, bv: TrgId, bw: TrgId },
-}
-
-/// Part of the net to link to; either a wire or a port.
-///
-/// To store this compactly, we reuse the [`Red`] tag to indicate if this is a
-/// wire.
-#[derive(Clone)]
-pub struct Trg(pub(crate) Port);
-
-impl fmt::Debug for Trg {
-  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-    if self.is_wire() { self.clone().as_wire().fmt(f) } else { self.0.fmt(f) }
-  }
-}
-
-impl Trg {
-  /// Creates a `Trg` from a port.
-  #[inline(always)]
-  pub fn port(port: Port) -> Self {
-    Trg(port)
-  }
-
-  /// Creates a `Trg` from a wire.
-  #[inline(always)]
-  pub fn wire(wire: Wire) -> Self {
-    Trg(Port(wire.0 as u64))
-  }
-
-  #[inline(always)]
-  fn is_wire(&self) -> bool {
-    self.0.tag() == Red
-  }
-
-  #[inline(always)]
-  fn as_wire(self) -> Wire {
-    Wire(self.0.0 as _)
-  }
-
-  #[inline(always)]
-  fn as_port(self) -> Port {
-    self.0
-  }
-
-  /// Access the target port; if this trg is already a port, this does nothing,
-  /// but if it is a wire, it loads the target.
-  ///
-  /// The returned port is only normative if it is principal; if it is a var or
-  /// a redirect, it must not be used for linking, and the original `Trg` should
-  /// be used instead.
-  #[inline(always)]
-  pub fn target(&self) -> Port {
-    if self.is_wire() { self.clone().as_wire().load_target() } else { self.0.clone() }
-  }
-}
-
-/// An index to a [`Trg`] in an [`Instruction`]. These essentially serve the
-/// function of registers.
-///
-/// When compiled, each `TrgId` will be compiled to a variable.
-///
-/// When interpreted, the `TrgId` serves as an index into a vector.
-#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct TrgId {
-  /// Instead of storing the index directly, we store the byte offset, to save a
-  /// shift instruction when indexing into the `Trg` vector in interpreted mode.
-  ///
-  /// This is always `index * size_of::<Trg>()`.
-  byte_offset: usize,
-}
-
-impl TrgId {
-  pub fn new(index: usize) -> Self {
-    TrgId { byte_offset: index * size_of::<Trg>() }
-  }
-  pub fn index(&self) -> usize {
-    self.byte_offset / size_of::<Trg>()
-  }
-}
-
-impl fmt::Display for TrgId {
-  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-    write!(f, "t{}", self.index())
-  }
-}
+mod addr;
+mod allocator;
+mod def;
+mod instruction;
+mod interact;
+mod linker;
+mod net;
+mod node;
+mod parallel;
+mod port;
+mod wire;
+
+pub use addr::*;
+pub use allocator::*;
+pub use def::*;
+pub use instruction::*;
+pub use linker::*;
+pub use net::*;
+pub use node::*;
+pub use port::*;
```

Comment:
I'm not very happy with having a lot of wildcard imports. We are pretending that the contents of these files are in `run`, when they actually aren't. 

However, if this change is just for easier navigation, then it makes sense.

Reply:
IMO, we shouldn’t make the internal organization of `run` public api — we want to be able to change that for readability without breaking everyone — so the inner modules are private, and the wildcard reexports make everything declared within public.